diag_manager/include/fms_diag_fieldbuff_update.fh

```diff
@@ -0,0 +1,1336 @@
+
+        IF  (PRESENT (mask)) THEN
+          mask_ptr(1:size(mask,1),1:size(mask,2),1:size(mask,3),1:1) => mask
+        ENDIF
+
+        succeded = FMS_DIAG_FBCF_PNAME_ (ofield_cfg, ofield_index_cfg, field_ptr, sample, &
+          & ofb_ptr, bbounds, count_0d, mask_ptr, missvalue,  &
+          & l_start, l_end, err_msg,  err_msg_local)
+    END FUNCTION FMS_DIAG_FBCF_3D_PNAME_
+
+!> @brief This code will be used by the preprecessor to generate an implementation
+!! to the module procudure for the fieldbuff_copy_fieldvals interface.
+!! The function may set or add to the output field buffer (argument ofb) with the input
+!! field data array  (argument field)
+FUNCTION FMS_DIAG_FBCF_PNAME_ (ofield_cfg, ofield_index_cfg, field, sample, ofb, &
+   & bbounds, count_0d, mask,  missvalue,  &
+   & l_start, l_end, err_msg,  err_msg_local) result( succeded )
+      TYPE(fms_diag_outfield_type), INTENT(in) :: ofield_cfg   !< The ofield_cfg object
+      TYPE(fms_diag_outfield_index_type) , INTENT(in) :: ofield_index_cfg    !< The ofield_index_cfg object
+      FMS_DIAG_FBU_DATA_TYPE_ , CONTIGUOUS, DIMENSION(:,:,:,:), INTENT(in) :: field   !< The field value array.
+      INTEGER, INTENT(in) :: sample   !< index along the diurnal time axis
+      FMS_DIAG_FBU_DATA_TYPE_ , DIMENSION(:,:,:,:,:), INTENT(inout) :: ofb   !< The Output Field Buffer
+      TYPE(fms_diag_ibounds_type), INTENT(inout) :: bbounds
+      FMS_DIAG_FBU_DATA_TYPE_ , INTENT(inout) :: count_0d   !> Normally the member of the buffer of same name,
+      LOGICAL, CONTIGUOUS, DIMENSION(:,:,:,:), INTENT(in), OPTIONAL :: mask
+      FMS_DIAG_FBU_DATA_TYPE_ , INTENT(in) :: missvalue !> buffer may be set to this value where mask is false.
+      INTEGER, DIMENSION(3), INTENT(in)  :: l_start   !< local start indices on spatial axes for regional output
+      INTEGER, DIMENSION(3), INTENT(in)  :: l_end   !< local end indices on spatial for regional output
+      CHARACTER(len=*), INTENT(inout), OPTIONAL :: err_msg
+      CHARACTER(len=256), INTENT(inout) :: err_msg_local
+      LOGICAL :: succeded !> Return true iff errors are not encounterd.
+      !!
+      !!
+      !> The indecies copied directly from the ofield_index_cfg
+      INTEGER :: is, js, ks, ie, je, ke, hi, hj, f1, f2, f3, f4
+
+      CHARACTER(:), ALLOCATABLE  :: output_name !< A copy of same variable in ofield_cfg
+      CHARACTER(:), ALLOCATABLE  :: module_name !< A copy of same variable in ofield_cfg
+      LOGICAL :: need_compute !< A copy of same variable in ofield_cfg
+      LOGICAL :: reduced_k_range !< A copy of same variable in ofield_cfg
+      LOGICAL :: mask_present !< A copy of same variable in ofield_cfg
+      LOGICAL :: missvalue_present !< A copy of same variable in ofield_cfg
+
+      INTEGER :: ksr, ker !> Loop indeccies used in reduced_k_range calculations
+      !> Looping indecies, derived from ofield_index_cfg:
+      INTEGER :: i, j, k, i1, j1, k1
+
+
+      LOGICAL :: time_max, time_min, time_sum
```

Comment:
these variables have the same name as parameter variables.  Hopefully the variables form the other module aren't used here